### PR TITLE
Maap 206 handle failed sms

### PR DESCRIPTION
**Pull request by elena-vi**
## Context

When an sms fails to send because of validation errors, the API endpoint didn't send back an error

## Changes proposed in this pull request

- Added another validation check (same regex as gov notify use)
- Make the api return the errors 

## Link to Jira card

https://hackney.atlassian.net/browse/MAAP-206

## Things to check

- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] Environment variables have been updated
- [x] Accompanying front end PR needs to be merged in before this change: https://github.com/LBHackney-IT/LBH-IncomeCollection/pull/304

### CODE REVIEW

**Comment by elena-vi** [comment]
Numquam est iusto v

File: lib/hackney/notification/gov_notify_gateway.rb [hunk 1]
```diff
@@ -14,24 +14,24 @@ def initialize(sms_sender_id:, api_key:, send_live_communications:, test_phone_n
       end
 
       def send_text_message(phone_number:, template_id:, reference:, variables:)
-        responce = client.send_sms(
+        response = client.send_sms(
```
Comment (C-gyorfi) [inline lib/hackney/notification/gov_notify_gateway.rb:17]: well spotted :D

File: lib/hackney/notification/send_manual_sms.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module Notification
     class SendManualSms < BaseManualGateway
       def execute(username:, tenancy_ref:, template_id:, phone_number:, reference:, variables:)
         phone = Phonelib.parse(phone_number)
-        if phone.valid?
+        if phone.valid? && govnotify_valid?(phone.full_e164)
```
Comment (C-gyorfi) [inline lib/hackney/notification/send_manual_sms.rb:8]: Probably not harmful, but do we need the old `phone.valid?` check or `govnotify_valid` would be enough?

**Comment by C-gyorfi** [comment]
Looks like there is a flaky test caused by MySQL, sad(ill have a look tomorrow) this is not something that introduced in this PR so good to go!

**Comment by elena-vi** [comment]
I can fix the test, don't you worry

**Pull request merged**